### ISSUE TITLE
Program quits when deleting directory in cloud

### ISSUE THREAD

**Issue by IHateTheProgrammingClass** [abraunegg/onedrive#1600]
**Note:** Before submitting a bug report, please ensure you are running the latest 'onedrive' client as built from 'master' and by using the latest available DMD compiler. Refer to the readme on building the client for your system.

### Bug Report Details ###
**Describe the bug**

After a long list of item processing the program arrives at a directory deletion then simply quits.

**Application and Operating System Details:**
*   Provide your OS & version (CentOS 6.x, Ubuntu 18.x etc) and the output of: `uname -a` Arch Latest I guess, Linux `<devicename> 5.10.56-1-lts #1 SMP Wed, 04 Aug 2021 12:34:06 +0000 x68_64 GNU/Linux`
*   Are you using a headless system (no gui) or with a gui installed? GUI
*   OneDrive Account Type SharePoint
*   Did you build from source or install from a package? AUR
*   If you installed from source, what is your DMD or LDC compiler version: `dmd --version` or `ldmd2 --version`
*   OneDrive Application Version: Output of `onedrive --version` v2.4.12-10-gl3b3709
*   OneDrive Application Configuration: Output of `onedrive --display-config`
*   Provide the version of curl you are using: Output of `curl --version` 7.78.0
*   Is your configured 'sync_dir' a local directory or a network mount point? Local
*   If *not* local, provide all the mountpoints in your system: Output of: `mount`
*   What partition format type does your configured 'sync_dir' reside on? Output of: `lsblk -f` NTFS
*   Explain your entire configuration setup - is the OneDrive folder shared with any other system, shared with any other platform at the same time, is the OneDrive account you use shared across multiple systems / platforms / Operating Systems and in use at the same time

I sync the dir to an external drive. The directory is encrypted with Cryptomator.

**Note:** Please generate a full debug log whilst reproducing the issue as per [https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support](https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support) and email to support@mynas.com.au

**To Reproduce**
Steps to reproduce the behavior if not causing an application crash:
1.  Start --sync... -v --force
2.  Stops at deleting remote directory

If issue is replicated by a specific 'file' or 'path' please archive the file and path tree & email to support@mynas.com.au 

**Complete Verbose Log Output**
A clear and full log of the problem when running the application in the following manner (ie, not in monitor mode):
```bash
onedrive --synchronize --verbose <any of your other needed options>
```

Run the application in a separate terminal window or SSH session and provide the entire application output including the error & crash. When posing the logs, Please format log output to make it easier to read. See [https://guides.github.com/features/mastering-markdown/](https://guides.github.com/features/mastering-markdown/) for more details.

Application Log Output:
```bash
Verbose console log output goes here
```

**Screenshots**

**Additional context**

### Bug Report Checklist ###
*   [x] Detailed description
*   [x] Application and Operating System Details provided in full
*   [x] Reproduction steps (if applicable)
*   [] Verbose Log Output from your error
*   [] Debug Log generated and submitted

**Comment by abraunegg** [comment]
@IHateTheProgrammingClass 
A few of things here:
1. The use of '--force' is for when local deletes are occurring, and these trigger the warning about large deletes
2. You have provided no actual log file / debug file or anything to support your claimed issue
3. You have not provided / filled in all the required information that helps me replicate your environment if there is an issue
4. You are running an unsupported client version. Please upgrade your client, test|try again and if you encounter the same issue, open a new ticket and provide all the required information.

**Comment by IHateTheProgrammingClass** [comment]
@abraunegg 1. I tried doing it without --force, and it seems to work.
2. I tried, could not.
4. I rely on AUR, if I update, I will update with that. I wont chance screwing up an installation built by me.

**Comment by IHateTheProgrammingClass** [comment]
@abraunegg I updated the comments and posts and sent a debug log to a companies email (?) which I fail to understand what affiliation has with this project. I have came to some conclusions. After updating the client the problem is still there. Somehow It managed to get working yesterday after the debug log. So I suspected that somehow enabled it? But turns out nope, I tried it again. Also in the debug log everything works fine, I get uploads, but not in the regular --sync[...]. I will try again with --monitor next time. It's really weird.
